### ISSUE TITLE
Pedestrians get stuck

### ISSUE THREAD

**Issue by Erida-Bendo** [Bonifatius94/PySocialForce#13]
![pedestrianFlow](https://github.com/Bonifatius94/PySocialForce/assets/127302774/ae751854-44fc-41ce-8f55-267706b0604a)

Hi @Bonifatius94 , I am working on adapting your code for Grasshopper (Rhino plugin). The goal would be to show how people move in a space with defined obstacles for reaching their goal locations.

For that I already created a new function that takes pedestrian locations from grasshopper for the simulation. However once I leave the simulation run I notice some strange behaviour where some of the pedestrians gather around the penultimate waypoint. I looked into changing the config factors namely the agent radius and the repulsive forces factor but was not able to affect this behaviour.

Do you have any thoughts on why this happens and what could be done to fix it? Thank you

**Comment by Erida-Bendo** [comment]
I managed to solve the issue, I was assigning waypoints in the wrong way when populating the simulation. Thank you for your work and for making it public though!

**Comment by Bonifatius94** [comment]
Hey Erida. Thank you for your interest in the project.  

FYI, this project was part of my master thesis on training an autonomous robot to navigate crowded zones. In my case we modeled the campus of the University of Augsburg.

For the simulation with the robot, I added some augmentations (with pygame) to visualize the robot's perception, show the route, etc. We could implement similar things for your specific case. 

Feel free to ask any further questions or open more GitHub issues. I was planning to improve the API / UX anyways. Your input is welcome. 

Best wishes, Marco

**Comment by Erida-Bendo** [comment]
Hi Marco,

thank you very much for your reply :). Sofar I was able to add an initial function where I take pedestrian locations from a txt file and then populate the simulation with them. In the end I take the positions to rhino grasshopper and visualize them there. I forked your repository here https://github.com/Erida-Bendo/PySocialForce 

I had a small issue where I was increasing the agent_radius in the config file, but could not see big changes in the simulation. Is that the correct variable for controlling pedestrian size?

Thank you very much for your help and time.

Warmly, Erida
